File: tests/test_code/javascript/jwt_ssrf_tests.js
const axios = require('axios');
const jwt = require('jsonwebtoken');
const express = require('express');
const app = express();

app.get('/fetch', (req, res) => {
    const target = req.query.url;
    // VULNERABLE: SSRF via axios
    axios.get(target).then(response => res.send(response.data));
});

app.post('/verify', (req, res) => {
    const token = req.body.token;
    // VULNERABLE: INSECURE_JWT (none algorithm)
    jwt.verify(token, secret, { algorithms: ['none', 'HS256'] }, (err, decoded) => {
        res.send(decoded);
    });
});

File: tests/test_code/python/jwt_ssrf_tests.py
import httpx
import jwt
from flask import Flask, request

app = Flask(__name__)

@app.route('/proxy')
def proxy():
    url = request.args.get('url')
    # VULNERABLE: SSRF via httpx
    return httpx.get(url).text

@app.route('/auth')
def auth():
    token = request.headers.get('Authorization')
    # VULNERABLE: INSECURE_JWT (verify=False)
    payload = jwt.decode(token, verify=False)
    
    # VULNERABLE: INSECURE_JWT (none algorithm allowed)
    payload2 = jwt.decode(token, algorithms=['none', 'HS256'])
    return str(payload)
